diff --git a/ipfs_accelerate_py b/ipfs_accelerate_py
--- a/ipfs_accelerate_py
+++ b/ipfs_accelerate_py
@@ -1 +1 @@
-Subproject commit 0cc04ebb640c4c981cf4650016e096a73ab0e8c0
+Subproject commit 0cc04ebb640c4c981cf4650016e096a73ab0e8c0-dirty
diff --git a/ipfs_datasets_py b/ipfs_datasets_py
index 9059235..d144be6 160000
--- a/ipfs_datasets_py
+++ b/ipfs_datasets_py
@@ -1 +1 @@
-Subproject commit 9059235ee84c17cdc5c971cedaf332529a1e6235
+Subproject commit d144be65ffe4c6423e4e1c30cd692812607343eb-dirty
diff --git a/ipfs_kit_py b/ipfs_kit_py
index ad19bd0..f6a5743 160000
--- a/ipfs_kit_py
+++ b/ipfs_kit_py
@@ -1 +1 @@
-Subproject commit ad19bd0a548139c540bd78163b9bea8dd38bf6cb
+Subproject commit f6a574375febbcf9a46fcd24bbc7bc5cfb551de5-dirty
diff --git a/swissknife b/swissknife
index 5cec1a4..df11f08 160000
--- a/swissknife
+++ b/swissknife
@@ -1 +1 @@
-Subproject commit 5cec1a41432a072cf86a8318c9dcb9493b6db7c2
+Subproject commit df11f08fae17d35153e420fdcdc5b38d9f6b9a7f-dirty
